modules/apps/data-engine/data-engine-taglib/src/test/java/com/liferay/data/engine/taglib/internal/servlet/taglib/util/DataLayoutTaglibUtilTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.data.engine.taglib.internal.servlet.taglib.util;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.liferay.data.engine.rest.resource.v2_0.DataDefinitionResource;
+import com.liferay.portal.json.JSONFactoryImpl;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.Collections;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Marcela Cunha
+ */
+public class DataLayoutTaglibUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() throws Exception {
+		_objectMapper = new ObjectMapper() {
+			{
+				configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true);
+			}
+		};
+
+		ReflectionTestUtil.setFieldValue(
+			_dataLayoutTaglibUtil, "_dataLayoutTaglibUtil",
+			_dataLayoutTaglibUtil);
+
+		ReflectionTestUtil.setFieldValue(
+			_dataLayoutTaglibUtil, "_jsonFactory", new JSONFactoryImpl());
+
+		ReflectionTestUtil.setFieldValue(
+			_dataLayoutTaglibUtil, "_dataDefinitionResourceFactory",
+			_dataDefinitionResourceFactory);
+
+		ReflectionTestUtil.setFieldValue(
+			_dataLayoutTaglibUtil, "_portal", _portal);
+
+		_setUpDataDefinitionResource();
+	}
+
+	@Test
+	public void testGetFieldTypesJSONArrayWithSearchableFieldsDisabled()
+		throws Exception {
+
+		JSONArray actualResultJSONArray =
+			_dataLayoutTaglibUtil.getFieldTypesJSONArray(
+				_httpServletRequest, Collections.singleton(_JOURNAL), false);
+
+		Assert.assertEquals(
+			_objectMapper.readTree(
+				read(
+					"data-definition-field-types-journal-scope-searchable-" +
+						"fields-enabled.json")),
```

Comment:
Hey @marcelabc ,

Overall, your code looks good, I'm just a little confuse about why are you checking if the result is equals to "fields-**enabled**.json" in this method that handles the opposite "WithSearchableFields**Disabled**".

What do you think to change it to be more clear for other developers?

Let me know if I understood something wrong. Thanks!